tools/cndump/kvset_dump.c

```diff
@@ -0,0 +1,398 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ * Copyright (C) 2022 Micron Technology, Inc.  All rights reserved.
+ */
+
+#include <assert.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <hse/error/merr.h>
+#include <hse_ikvdb/diag_kvdb.h>
+#include <hse_ikvdb/omf_kmd.h>
+#include <hse_util/parse_num.h>
+#include <hse_util/fmt.h>
+
+#include <mpool/mpool_structs.h>
+#include <mpool/mpool.h>
+
+#include <cn/omf.h>
+
+#include "cndb_reader.h"
+#include "cndb_record.h"
+#include "fatal.h"
+#include "globals.h"
+
+struct opt {
+    uint64_t kvset_id;
+};
+
+static struct opt opt;
+
+struct dump_mblock {
```

Comment:
This struct name sounds like a function name to me, maybe it's just me...

Reply:
The great thing about C namespaces is that you could have something like:

int dump_mblock(struct dump_mblock *p, int dump_mblock);

IIRC...

Reply:
gonna go wash out my eyes real quick

Reply:
leaving as is.  it's local to a file so name doesn't matter too much :)

Reply:
This struct seems to represent an `mblock dump`, so `struct mblock_dump` makes more sense to me, but do whatever you want.